File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
'''
[문제 분석]
- 배열 arr 이 주어질 때,(0~9까지 이루어져 있다.)
- 연속적으로 나타나는 숫자는 1쌍 빼고 다 삭제해야 한다. 
- 제거된 후 남은 수를 반환할 때 원소들의 순서를 유지해야 한다. 

> 조건 
> 입력값 
> 출력값
[해결 논리]
- i를 계속 이동해가면서 i+1 이 i 위치의 값과 같으면 한개만 남겨야 한다. 
- 왼쪽에서 오른쪽 방향으로 탐색해야한다. 

'''
def solution(arr):
    answer = []
    n = len(arr)
    for i in range(n):
        # 마지막 원소거나, 다음 원소와 다르면 append 
        if i == (n-1) or arr[i] != arr[i+1]:
            answer.append(arr[i])
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answerList = new ArrayList<>();
        int prev = -1; // arr 값은 0~9, 안전하게 초기화
        
        for(int num : arr){
            if(num != prev){
                answerList.add(num);
                prev = num;
            }
        }
        
        int[] answer = new int[answerList.size()];
        for(int i = 0; i < answerList.size(); i++){
            answer[i] = answerList.get(i);
        }
        return answer;
    }
}
